gettext-rs/src/macros.rs

```diff
@@ -1,50 +1,85 @@
+#[macro_export]
+macro_rules! freplace {
+    ($input:expr, $($args:tt)*) => {{
+        let mut parts = $input.split("{}");
+        let mut output = parts.next().unwrap_or_default().to_string();
+        $(
+            output += &format!("{}{}", $args, &parts.next().unwrap());
```

Comment:
What happens if the number of arguments doesn't match the number of placeholders? Do we get the dreaded error that says that the panic comes from a macro and we need Rust Nightly and some -Z flag to figure out what's going on?

Reply:
Ah yes, probably. I suppose we should use `expect` and a meaningful error message here to make it less painful. I'd still keep the panic since supplying a different number of arguments is a programmer error though.

Reply:
Absolutely! I was only concerned about the quality of error reporting.